modules/core/src/edu/kit/iti/algover/term/prettyprint/MultiSubscriptPrinterExtension.java

```diff
@@ -0,0 +1,78 @@
+package edu.kit.iti.algover.term.prettyprint;
+
+import edu.kit.iti.algover.term.ApplTerm;
+import edu.kit.iti.algover.term.FunctionSymbol;
+import edu.kit.iti.algover.term.VariableTerm;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MultiSubscriptPrinterExtension implements PrettyPrintExtension, VariablePrettyPrintExtension {
+
+    private static final String SUBSCRIPT_PATTERN = "(.*)_([0-9]+)";
```

Comment:
This pattern does not look right. Are there test cases for this pretty printing?

Reply:
Pattern is taken form existing SubscriptPrinterExtension and applied "recursively" on group 1 until there is no match. Maybe there is a more elegant way to match all indices. 
There is a testMultiLevelIndexing() method in the PrettyPrintTest class.

```